src/components/UserProfile/FoodTicket.tsx

```diff
@@ -0,0 +1,144 @@
+"use client";
+
+import { generateClient } from "aws-amplify/api";
+import Image from "next/image";
+import { QRCode } from "react-qrcode-logo";
+
+import { type Schema } from "@/amplify/data/resource";
+import { createMessageAndCode } from "@/amplify/function/BusinessLogic/utils/crytography";
+import { getUpcomingFoodEventDetails } from "@/app/get-food-ticket/actions";
```

Comment:
Move any actions into the components folder, either under user profile or somewhere you think makes sense. They should not be in the app folder, this was only for demoing. Also delete the sample.tsx there since you have now re-implemented it here.(that was also just a sample)

Reply:
@ideen1 I moved the action into src/components/actions and then deleted the sample.tsx related to the action. I didnt delete the others ones (ex: scan-food-ticket) because that hasnt been implemented yet if thats what you mean

Reply:
Once the scan food ticket is implemented on the admin side then that action can also be moved into the actions folder and the other sample can be deleted